tool/planet/constants.go

```diff
@@ -542,43 +544,21 @@ var KubeletVersion = schema.GroupVersionKind{
 	Kind:    "KubeletConfiguration",
 }
 
-var allCaps = []string{
-	"CAP_AUDIT_CONTROL",
-	"CAP_AUDIT_READ",
-	"CAP_AUDIT_WRITE",
-	"CAP_BLOCK_SUSPEND",
-	"CAP_CHOWN",
-	"CAP_DAC_OVERRIDE",
-	"CAP_DAC_READ_SEARCH",
-	"CAP_FOWNER",
-	"CAP_FSETID",
-	"CAP_IPC_LOCK",
-	"CAP_IPC_OWNER",
-	"CAP_KILL",
-	"CAP_LEASE",
-	"CAP_LINUX_IMMUTABLE",
-	"CAP_MAC_ADMIN",
-	"CAP_MAC_OVERRIDE",
-	"CAP_MKNOD",
-	"CAP_NET_ADMIN",
-	"CAP_NET_BIND_SERVICE",
-	"CAP_NET_BROADCAST",
-	"CAP_NET_RAW",
-	"CAP_SETGID",
-	"CAP_SETFCAP",
-	"CAP_SETPCAP",
-	"CAP_SETUID",
-	"CAP_SYS_ADMIN",
-	"CAP_SYS_BOOT",
-	"CAP_SYS_CHROOT",
-	"CAP_SYS_MODULE",
-	"CAP_SYS_NICE",
-	"CAP_SYS_PACCT",
-	"CAP_SYS_PTRACE",
-	"CAP_SYS_RAWIO",
-	"CAP_SYS_RESOURCE",
-	"CAP_SYS_TIME",
-	"CAP_SYS_TTY_CONFIG",
-	"CAP_SYSLOG",
-	"CAP_WAKE_ALARM",
+var allCaps = []string{}
+
+// Based on runc capabilities detection
+// https://github.com/opencontainers/runc/blob/2e94378464ae22b92e1335c200edb37ebc94a1b7/libcontainer/capabilities_linux.go#L17-L31
+func init() {
+	last := capability.CAP_LAST_CAP
+	// workaround for RHEL6 which has no /proc/sys/kernel/cap_last_cap
+	if last == capability.Cap(63) {
```

Comment:
This feels like a sub-optimal way of detecting RHEL6. What if there are 64 capabilities supported? I think it would be safer to probe down from `capabilities.CAP_LAST_CAP` until one is found that works and set that as the last for the case it equals `63` initially - i.e. either it has not been read from cap_last_cap file (as the case is with some RHEL6 distributions) or it has read `63` (in which case it might not be RHEL6) - what do you think?

Reply:
I think we don't support RHEL6 so it's not really worth worrying about, unless there is some evidence cap_last_cap is also missing in RHEL 7.4+. This was literally just a direct copy of the logic from libcontainer but ported to be a list instead of a map which planet is already using.